File: src/main/java/io/beaniejoy/dongnecafe/cafe/dto/cafe/CafeInfoResponseDto.java
package io.beaniejoy.dongnecafe.cafe.dto.cafe;

import io.beaniejoy.dongnecafe.cafe.dto.menu.CafeMenuListResponseDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CafeInfoResponseDto {
    private UUID cafeId;

    private String name;

    private String address;

    private String phoneNumber;

    private Double totalRate;

    private String description;

    private List<CafeMenuListResponseDto> menuList;

    private List<CafeImageResponseDto> imageList;
}

File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/dto/cafe/CafeInfoResponseDto.kt
package io.beaniejoy.dongnecafe.domain.cafe.dto.cafe

import io.beaniejoy.dongnecafe.domain.cafe.entity.Cafe
import io.beaniejoy.dongnecafe.domain.cafe.dto.menu.CafeMenuResponseDto

data class CafeInfoResponseDto(
    val id: Long? = null,
    val name: String? = null,
    val address: String? = null,
    val phoneNumber: String? = null,
    val totalRate: Double? = null,
    val description: String? = null,
    val menuList: List<CafeMenuResponseDto> = emptyList(),
    val imageList: List<CafeImageResponseDto> = emptyList()
) {
    companion object {
        fun of(cafe: Cafe): CafeInfoResponseDto {
            return CafeInfoResponseDto(
                id = cafe.id,
                name = cafe.name,
                address = cafe.address,
                phoneNumber = cafe.phoneNumber,
                totalRate = cafe.totalRate,
                description = cafe.description,
                menuList = cafe.cafeMenuList.map { CafeMenuResponseDto.of(it) },
                imageList = cafe.cafeImageList.map { CafeImageResponseDto.of(it) }
            )
        }
    }
}